Hippo.Core/Services/PuppetService.cs

```diff
@@ -17,23 +19,55 @@ public interface IPuppetService
     public class PuppetService : IPuppetService
     {
         private readonly PuppetSettings _settings;
-        private readonly GitHubClient _gitHubClient;
+        private readonly IMemoryCache _memoryCache;
 
-        public PuppetService(IOptions<PuppetSettings> settings)
+        public PuppetService(IOptions<PuppetSettings> settings, IMemoryCache memoryCache)
         {
             _settings = settings.Value;
-            _gitHubClient = new GitHubClient(new ProductHeaderValue("Hippo"))
+            _memoryCache = memoryCache;
+        }
+
+        private async Task<GitHubClient> GetGithubClient()
+        {
+            // caching token for 10 minutes, which is the limit for GitHub API's
+            if (!_memoryCache.TryGetValue("github-app-installation-token", out string appInstallationToken))
```

Comment:
usually it's safer or at least more common to use `GetOrCreateAsync<T>` when working with memory cache since it'll handle the missing data case and wraps it nicely so you don't have to worry about race conditions, plus just more obvious what it is trying to do

Reply:
GetOrCreate doesn't appear to have an override that takes `MemoryCacheEntryOptions` arg.

Reply:
Oh, I see. The delegate provides a `ICacheEntry` arg. Okay.